test/model.test.js

```diff
@@ -0,0 +1,893 @@
+    .then(function() {
+      thinky = test.thinky;
+      type = thinky.type;
+    });
+});
+after(function() { return test.dropDatabase(); });
+
+describe('createModel', function() {
+  afterEach(function() { return test.dropTables(); });
+
+  it('should create a new model', function() {
+    var model = thinky.createModel(util.s8(), {
+      id: String, name: String
+    });
+
+    return model.ready()
+      .then(function() { expect(model).to.exist; });
+  });
+
+  it('should check if the table was created', function() {
+    var modelName = util.s8();
+    var model = thinky.createModel(modelName, {
+      id: String, name: String
+    });
+
+    return model.ready()
+      .then(function() {
+        return test.r.tableList().run();
+      })
+      .then(function(result) {
+        expect(result).to.contain(modelName);
+      });
+  });
+
+  it('should create multiple models', function() {
+    var model1 = thinky.createModel(util.s8(), {
+      id: String, name: String
+    });
+
+    var model2 = thinky.createModel(util.s8(), {
+      id: String, name: String
+    });
+
+    return Promise.all([ model1.ready(), model2.ready() ])
+      .then(function() {
+        expect(model1).to.not.eql(model2);
+      });
+  });
+
+  it('should throw error for non existent table access', function() {
+    var model = thinky.createModel("nonExistentTable", {
+      id: String, name: String
+    }, {
+      init: false
+    });
+
+    return expect(model.get(1).run())
+      .to.eventually.be.rejectedWith(/^Table `.*` does not exist in/);
+  });
+});
+
+describe('[_]getModel', function() {
+  it('_getModel', function() {
+    var name = util.s8();
+    var model = thinky.createModel(util.s8(), {
+      id: String, name: String
+    }, {
+      init: false
+    });
+
+    expect(model._getModel()).to.have.property('_name');
+  });
+
+  it('getTableName', function() {
+    var modelName = util.s8();
+    var model = thinky.createModel(modelName, {
+      id: String, name: String
+    }, {
+      init: false
+    });
+
+    expect(model).to.have.property('getTableName');
+    expect(model.getTableName()).to.equal(modelName);
+  });
+});
+
+describe('Model', function() {
+  before(function() {
+    test.models.Model = thinky.createModel(util.s8(), { str: String });
+    return test.models.Model.ready();
+  });
+  after(function() { return test.dropTables(); });
+  afterEach(function() { return test.clearTables(); });
+
+  it('should create a new instance of a model', function() {
+    var str = util.s8();
+    var doc = new test.models.Model({ str: str });
+
+    expect(util.isPlainObject(doc)).to.be.true;
+    expect(doc.str).to.eql(str);
+  });
+
+  it('should create multiple instances from the same document', function() {
+    var str = util.s8();
+    var num = util.random();
+
+    var values = { str: str, num: num };
+    var doc = new test.models.Model(values);
+    var otherDoc = new test.models.Model(values);
+
+    expect(doc).to.eql(values);
```

Comment:
This is different than testing `doc === values`.
It's actually an unusual equality test from what I could test.

My guess is without looking in the code, it does something like this
```
for(key in doc) {
  if (doc[key] != values[key]) { // they probably recurse here instead of checking !=
    throw ...
  }
}
```

Meaning that
```
var a = {foo: "bar"}
var b = {};
b.__proto__.foo = "bar"
expect(a).to.eql(b) // doesn't throw...
```